File: TowerDefence.spritebuilder/Unit/Tower/TargetFinder.swift
//
//  TargetFinder.swift
//  TowerDefence
//
//  Created by koveres on 07.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation


public class TargetFinder {
    
    var solderEngine:SolderEngine?;
    var solders:[Solder]?;
    
    public init( solders1:[Solder]) {
     solders = solders1;
    }
    public init( solderEngine1:SolderEngine) {
     solderEngine = solderEngine1;
    }
    
 
    
    
    public func getSolder(  center:CGPoint,   distance:CDouble) -> ([Solder]) {
    if(solderEngine != nil) {
        solders = solderEngine!.GetSoldersOnMap();
    }
    
     var solderList:[Solder]  = [Solder]();
    for  solder in solders!
    {
        var centerSolder:CGPoint? =   solder.getCurrentState();
        var distanceToSolder:CGFloat = ccpDistance(centerSolder!, center);
    if(CDouble(distanceToSolder) <  distance ) {
    if(!solder.getIsDead()){
    if(!solder.isGoOut()){
        solderList.append(solder);}}
    }
    
    }
    return solderList;
    }
    
}

File: test.spritebuilder/Source/TDEngine/Unit/Tower/TargetFinder.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Unit.Tower;

import Engine.SolderEngine;
import Unit.Solder.Solder;
import Utils.Point2D;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author vereshchagin
 */
public class TargetFinder {

    SolderEngine solderEngine;
    public TargetFinder(List<Solder> solders) {
        this.solders = solders;
    }
    public TargetFinder(SolderEngine solderEngine) {
        this.solderEngine = solderEngine;
    }
    
    List<Solder> solders;
    
    
    public List<Solder> getSolder(Point2D center, double distance) {
        if(solderEngine != null) {
            solders = solderEngine.GetSoldersOnMap();
        }
        
        List<Solder> solderList = new ArrayList<Solder>();
        for (Solder solder : solders) {
          Point2D centerSolder =   solder.getCurrentState();
          double distanceToSolder = centerSolder.distance(center);
          if(distanceToSolder< distance) {
             if(!solder.getIsDead()) 
                if(!solder.isGoOut())  
                   solderList.add(solder);
            }
            
        }
        return solderList;
    }
    
}
